config/wp-config-template.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * The base configuration for WordPress.
+ * This file will be copied into /public by Composer after WordPress is installed.
+ *
+ * @package Dekode
+ */
+
+declare( strict_types = 1 );
+
+/* Application root directory. */
+$app_root = dirname( __DIR__ );
+
+/* Autoload Composer packages. */
+require_once $app_root . '/vendor/autoload.php';
+
+/* Load environment variables */
+if ( file_exists( $app_root . '/.env' ) ) {
+	$dotenv = new \Symfony\Component\Dotenv\Dotenv();
+	$dotenv->load( $app_root . '/.env' );
+}
+
+/**
+ * Get env variable with empty fallback.
+ *
+ * @param string $key Variable key.
+ * @param mixed  $default Default value if key is not set.
+ * @return mixed
+ */
+function env( string $key, $default = '' ) { // phpcs:ignore NeutronStandard.Functions.TypeHint.NoArgumentType, NeutronStandard.Functions.TypeHint.NoReturnType
+	$value = $_ENV[ $key ] ?? $default;
+
+	// Return bool value for 'true' or 'false'.
+	if ( in_array( $value, [ 'true', 'false' ], true ) ) {
+		return boolval( $value );
+	}
+
+	return $value;
+}
+
+/**
+ * Conditionally use, or generate, `WP_HOME` and `WP_SITEURL`.
+ */
+if ( env( 'WP_HOME' ) ) {
+	define( 'WP_HOME', env( 'WP_HOME' ) );
+	define( 'WP_SITEURL', env( 'WP_SITEURL', WP_HOME ) );
+} else {
+	$http_host   = filter_input( INPUT_SERVER, 'HTTP_HOST', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+	$server_port = filter_input( INPUT_SERVER, 'SERVER_PORT', FILTER_SANITIZE_NUMBER_INT );
+	$http_x_fp   = filter_input( INPUT_SERVER, 'HTTP_X_FORWARDED_PROTO', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+	$https       = filter_input( INPUT_SERVER, 'HTTPS', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+	$scheme      = 'http';
+
+	if ( ( is_string( $https ) && 'on' === strtolower( $https ) ) || '443' === $server_port || 'https' === $http_x_fp ) {
+		$scheme           = 'https';
+		$_SERVER['HTTPS'] = 'on';
+	}
+
+	define( 'WP_HOME', $scheme . '://' . $http_host );
+	define( 'WP_SITEURL', WP_HOME );
```

Comment:
Should WP_HOME be required? Then we can skip this section and die if WP_HOME is missing from .env

Reply:
I'd like it to be, yes. The idea behind getting the server name from the server it self can lead to issues where the first site on a server is the default, leading to potential duplicate content indexing and similar, so let's work against that behavior by enforcing the use of `WP_HOME` again.

Reply:
That is only a problem when using `$_SERVER['SERVER_NAME'];`. When using `HTTP_HOST` as above, it is not a problem as only the currently used host header will be used as `WP_HOME`.

I don't think `WP_HOME` should be required in `.env`, unless the logic for `WP_CONTENT_URL` is changed. Ie. for multisite networks where a subsite is living on reisetips.nettavisen.no, it will still insist on loading assets from `WP_HOME` (reisetips.prod.dekodes.no). Unsetting `WP_HOME` in `.env` for reisetips was the solution to this problem.

On the contrary, relying on a 100% dynamic `WP_HOME` is also bad, since this moves the responsibility of redirections entirely to the webserver since Wordpress now accepts any host header, whereas with a defined `WP_HOME` it has something to fall back on. Many existing clients have domains added to the web server config, without any redirects, relying on Wordpress to 301 alt domains to the main domain.

Thus, I think the current behaviour of using the defined `WP_HOME` if defined and dynamical if not is the way to go.

Reply:
Tnx for the input, @Clorith and @dekode-andre. Let's keep this as-is for now and look further into this at a later stage.